### PR TITLE
d_do_test: Don't link with OPT:NOICF by default on windows

### PR DESCRIPTION
**Pull request by MoonlightSentinel**
The issue related to COMDAT folding was fixed in #11736.

### CODE REVIEW

**Comment by rainers** [comment]
Thanks for tackling this. While it might work for the given case, I'm not sure this is the proper fix. Can't the same issue happen with stack unwinding?

AFAICT this adds an unused symbol reference to the catch clause. I'd rather expect that the exception handling data "associated" with the function has to be compared aswell for functions to be merged. I suspect there is a weakrer coupling emitted by dmd than by the C++ compiler.

If this is considered good enough, please also remove the linker option from ini/.../sc.ini and the phobos win64.mak.

File: test/tools/d_do_test.d [hunk 1]
```diff
@@ -452,11 +452,6 @@ bool gatherTestParameters(ref TestArgs testArgs, string input_dir, string input_
     if (testArgs.mode == TestMode.FAIL_COMPILE)
         testArgs.requiredArgs = "-verrors=0 " ~ testArgs.requiredArgs;
 
-    // https://issues.dlang.org/show_bug.cgi?id=10664: exceptions don't work reliably with COMDAT folding
-    // it also slows down some tests drastically, e.g. runnable/test17338.d
```
Comment (PetarKirov) [inline test/tools/d_do_test.d:456]: What about the show down mentioned by this comment. Is it now fixed?

Comment (MoonlightSentinel) [reply test/tools/d_do_test.d:456]: The slowdown is on average slightly less than 0.1 seconds per test (except for a few tests with many permutations).

`runnable/test17338.d` was the only exception with more than 80-fold slowdown, therefore `-L/OPT:NOICF` is now explicitly specified as a `REQUIRED_ARGS`.

File: test/runnable/test17338.d [hunk 2]
```diff
@@ -1,5 +1,8 @@
 // PERMUTE_ARGS:
 
+// COMDAT folding increases runtime by > 80x
+// REQUIRED_ARGS(windows): -L/OPT:NOICF
```
Comment (rainers) [inline test/runnable/test17338.d:4]: optlink will probably print a message when testing OMF builds with this option, but it doesn't seem critical.

**Review by rainers — APPROVED** [review]

**Pull request merged**